### DISCUSSION TITLE
QueryEntities Unlimited Components

### DISCUSSION THREAD

**Discussion by simeonradivoev** [sebas77/Svelto.ECS#81] [Q&A] [answered]
How do you go about querying more than 4 component types from the generic method QueryEntities?
I can see there is a mapper, but it doesn't feel like its intended use.

**Comment by sebas77** [comment, marked as answer]
as long as all the entities in the groups you are querying have the components you need, they can all be indexed with the same index.
For this reason, you can call QueryEntities how many times you need.
With some Svelto patterns you instead get the current group from the foreach and use it to do the extra query entities inside the foreach.

**Comment by simeonradivoev** [comment]
What if the group has entities with different descriptors?

**Comment by sebas77** [comment]
Svelto patterns advise using 1 EntityDescriptor = 1 group. If you mix entity descriptors in the same group, it's an advanced use and you must know what you are doing.
Now as long as the entity descriptors you are mixing have the same components queried shared is still not a problem.
If they don't, then you are forced to use the egidmapper.
